app/content/views/user.py

```diff
@@ -137,12 +139,55 @@ def get_user_memberships(self, request, *args, **kwargs):
         serializer = GroupSerializer(groups, many=True, context={"request": request})
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    @action(detail=False, methods=["get"], url_path="me/badges")
-    def get_user_badges(self, request, *args, **kwargs):
-        user_badges = request.user.user_badges.order_by("-created_at")
-        badges = [user_badge.badge for user_badge in user_badges]
+    def post_user_badges(self, request, *args, **kwargs):
+
+        user = self.request.user
+        badge = get_object_or_404(Badge, flag=request.data.get("flag"))
+
+        if not badge.active:
+            return Response(
+                {"detail": "Badgen er ikke aktiv"}, status=status.HTTP_400_BAD_REQUEST,
+            )
+        serializer = UserBadgeSerializer(data=request.data)
+        if serializer.is_valid():
+            UserBadge(user=user, badge=badge).save()
```

Comment:
Skal forklare på møte hvordan logging nå skal gjøres, tror det følgende er mulig men hvis ikke så kan du logge manuelt
```suggestion
            super().perform_create(serializer,user=user, badge=badge)
            # Evt:
            self._log_on_create(serializer)
```